Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -32,33 +32,46 @@ public protocol ConfigurableRule: Rule {
     func isEqualTo(rule: ConfigurableRule) -> Bool
 }
 
-public protocol ParameterizedRule: ConfigurableRule {
-    typealias ParameterType: Equatable
-    init(parameters: [RuleParameter<ParameterType>])
-    var parameters: [RuleParameter<ParameterType>] { get }
+public protocol ViolationLevelRule: ConfigurableRule {
+    var warning: RuleParameter<Int> { get set }
+    var error: RuleParameter<Int> { get set }
 }
 
-// Default implementation for ConfigurableRule conformance
-extension ParameterizedRule {
+public protocol CorrectableRule: Rule {
+    func correctFile(file: File) -> [Correction]
+}
+
+// MARK: - ViolationLevelRule conformance to ConfigurableRule
+
+public extension ViolationLevelRule {
     public init?(config: AnyObject) {
-        guard let array = [ParameterType].arrayOf(config) else {
+        self.init()
+        if let config = [Int].arrayOf(config) where !config.isEmpty {
+            warning = RuleParameter(severity: .Warning, value: config[0])
+            if config.count > 1 {
+                error = RuleParameter(severity: .Error, value: config[1])
+            }
+        } else if let config = config as? [String: AnyObject] {
+            if let warningNumber = config["warning"] as? Int {
+                warning = RuleParameter(severity: .Warning, value: warningNumber)
+            }
+            if let errorNumber = config["error"] as? Int {
+                error = RuleParameter(severity: .Error, value: errorNumber)
+            }
+        } else {
             return nil
         }
-        self.init(parameters: RuleParameter<ParameterType>.ruleParametersFromArray(array))
```

Comment:
I don't think `ruleParametersFromArray` is used anymore at all, can you please remove it?